packages/server/src/course/course.controller.ts

```diff
@@ -26,9 +27,46 @@ export class CourseController {
 
   @Get(':id')
   async get(@Param('id') id: number): Promise<GetCourseResponse> {
-    return await CourseModel.findOne(id, {
-      relations: ['officeHours', 'queues', 'queues.staffList'],
+    // TODO: for all course endpoint, check if they're a student or a TA
+    const course = await CourseModel.findOne(id, {
+      relations: ['officeHours'],
     });
+
+    const now = new Date();
+    const MS_IN_MINUTE = 60000;
+
+    const rooms = course.officeHours
+      .filter(
+        (e) =>
+          e.startTime.valueOf() - 15 * MS_IN_MINUTE < now.valueOf() &&
+          e.endTime.valueOf() + 1 * MS_IN_MINUTE > now.valueOf(),
+      )
+      .map((e) => e.room);
+
+    const queues = await QueueModel.find({
+      where: {
+        courseId: id,
+      },
+      relations: ['staffList'],
+    });
+
+    const nonEmptyQueues = queues.filter((e) => e.staffList.length > 0);
+
+    const queuesHappeningNow = [];
+
+    for (const q of queues) {
+      const oh = course.officeHours.find((f) => f.room === q.room);
```

Comment:
Should this line reference the `rooms` const from before?
```suggestion
      const oh = rooms.find((f) => f.room === q.room);
```